judge/utils/problems.py

```diff
@@ -44,11 +44,7 @@ def contest_attempted_ids(participation):
     key = 'contest_attempted:%s' % participation.id
     result = cache.get(key)
     if result is None:
-        result = {id: {'achieved_points': points, 'max_points': max_points}
-                  for id, max_points, points in (participation.submissions
-                                                 .values_list('problem__problem__id', 'problem__points')
-                                                 .annotate(points=Max('points'))
-                                                 .filter(points__lt=F('problem__points')))}
+        result = set(participation.submissions.values_list('problem__problem__id', flat=True).distinct())
```

Comment:
Does this work? Would save a join.

```suggestion
        result = set(participation.submissions.values_list('problem__problem_id', flat=True).distinct())
```

Reply:
Yes, `problem_id` exists. Done.